src/Store.js

```diff
@@ -3,35 +3,22 @@
  * Used to provide default values for a store configuration
  */
 
-let attempt = require('./attempt')
-let tag     = require('./tag')
-
-let Store = {
-
-  getInitialState(store) {
-    return attempt(store, 'getInitialState')
-  },
-
-  serialize(store, state) {
-    return attempt(store, 'serialize', [ state ], state)
-  },
-
-  deserialize(store, state) {
-    return attempt(store, 'deserialize', [ state ], state)
-  },
-
-  register(store) {
-    return attempt(store, 'register', [], undefined, store)
-  },
+exports.getInitialState = function (store) {
+  if ('getInitialState' in store) {
+    return store.getInitialState()
+  }
+}
 
-  send(store, state, transaction) {
-    let { action, body } = transaction
+exports.serialize = function (store, state) {
+  return 'serialize' in store ? store.serialize(state) : state
+}
 
-    tag(action)
+exports.deserialize = function (store, raw) {
+  return 'deserialize' in store ? store.deserialize(raw) : raw
+}
 
-    return attempt(Store.register(store), action.toString(), [ state, body ], state, store)
-  }
+exports.send = function (store, state, { payload, type }) {
+  let handler = 'register' in store ? store.register()[type] : false
 
+  return handler ? handler.call(store, state, payload) : state
 }
-
-module.exports = Store
```

Comment:
I dropped `attempt`. It was super hard to parse and, while this is verbose, I like the clarity